swap/peer.go

```diff
@@ -30,18 +34,142 @@ var ErrDontOwe = errors.New("no negative balance")
 // Peer is a devp2p peer for the Swap protocol
 type Peer struct {
 	*protocols.Peer
+	lock               sync.RWMutex
 	swap               *Swap
 	beneficiary        common.Address
 	contractAddress    common.Address
 	lastReceivedCheque *Cheque
+	lastSentCheque     *Cheque
+	balance            int64
 }
 
 // NewPeer creates a new swap Peer instance
-func NewPeer(p *protocols.Peer, s *Swap, beneficiary common.Address, contractAddress common.Address) *Peer {
-	return &Peer{
+func NewPeer(p *protocols.Peer, s *Swap, beneficiary common.Address, contractAddress common.Address) (*Peer, error) {
+	peer := &Peer{
 		Peer:            p,
 		swap:            s,
 		beneficiary:     beneficiary,
 		contractAddress: contractAddress,
 	}
+
+	var err error
```

Comment:
You could initialize this in the function initialization

Reply:
done

Reply:
I see what you did, sorry for not being clear (please see my [other comment](https://github.com/ethersphere/swarm/pull/1725#discussion_r322674276)), for a better explanation

Reply:
Changed